### DISCUSSION TITLE
Loading the models from local path while using ocr_predictor()

### DISCUSSION THREAD

**Discussion by ZibaGandomkar** [mindee/doctr#1238] [Q&A] [answered]
I want to package an app using ocr_predictor for users without access to internet (due to some settings they cannot download anything while running my app). I came across this post: [(https://github.com/mindee/doctr/discussions/1151)](https://github.com/mindee/doctr/discussions/1151). So, I assume something like the following code should work. However, although for "crnn_vgg16_bn", pretrained is False, it still downloads the model. Any help is highly appreciated. 

import torch
from doctr.models import ocr_predictor
from doctr.models import db_resnet50, crnn_vgg16_bn

detection_model_path=r'path_to_folder\db_resnet50-ac60cadc.pt'
recognition_model_path = r'path_to_folder\vgg16_bn_r-d108c19c.pt'

det_model = db_resnet50(pretrained=False)
det_params = torch.load(detection_model_path, map_location="cpu")
det_model.load_state_dict(det_params)

# Load recognition model
reco_model = crnn_vgg16_bn(pretrained=False)
reco_params = torch.load(recognition_model_path, map_location="cpu")
reco_model.load_state_dict(reco_params)

#predictor
predictor = ocr_predictor(det_model, reco_model, pretrained=True)
print(predictor)

**Comment by felixdittrich92** [comment]
Hi @ZibaGandomkar :wave:,

`predictor = ocr_predictor(det_model, reco_model, pretrained=True)` will still download the default models without using them if you pass `pretrained=True`.

**Reply by ZibaGandomkar** [reply]
Hi @felixdittrich92 

Thanks for your response. I've tried predictor = ocr_predictor(det_model, reco_model, pretrained=False), but still it downloaded the reco_model. Then I noted the problem is with "reco_model = crnn_vgg16_bn(pretrained=False)". Although the pretrained flag is False, it still downloads the model. I'm wondering if you know why it's happening. Many thanks!

**Reply by ZibaGandomkar** [reply]
Hi again @felixdittrich92, Problem solved :) For future users: the _pretrained_backbone_ false should be also explicitly set to "False".
reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)

**Comment by ZibaGandomkar** [comment, marked as answer]
(1) Set the _pretrained_backbone_ to **False**: reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
(2) Set _pretrained_ flag to **False** in predictor = ocr_predictor(det_model, reco_model, pretrained=False)
